sdks/python/setup.py

```diff
@@ -139,7 +139,7 @@ def get_version():
     # version of dill. It is best to use the same version of dill on client and
     # server, therefore list of allowed versions is very narrow.
     # See: https://github.com/uqfoundation/dill/issues/341.
-    'dill>=0.3.1.1,<0.3.2',
+    'dill>=0.3.1.1,<0.3.4',
```

Comment:
we need to keep a very tight upper + lower bound dill, see: https://lists.apache.org/thread.html/r90940a6953f546f4839ade82dfad2e494c4273a1931505ddebcb7627%40%3Cdev.beam.apache.org%3E

Reply:
so you'd have to write 

>=0.3.4,<0.3.5

Reply:
most likely this is what is causing test failures.

Reply:
Also minor correction to note: the current bound <0.3.4 does not include 3.4

Reply:
thanks a lot for looking into these changes though, dill is a tricky dependency to deal with...

Reply:
cc: @ryanthompson591